test/e2e/statefulset_arbitrary_config/statefulset_arbitrary_config_test.go

```diff
@@ -50,4 +67,20 @@ func TestStatefulSetArbitraryConfig(t *testing.T) {
 	t.Run("Container has been merged by name", mongodbtests.StatefulSetContainerConditionIsTrue(&mdb, "mongodb-agent", func(container corev1.Container) bool {
 		return container.ReadinessProbe.TimeoutSeconds == 100
 	}))
+	t.Run("Tolerations have been added correctly", mongodbtests.StatefulSetConditionIsTrue(&mdb, func(sts appsv1.StatefulSet) bool {
```

Comment:
It seems that this test doesn't check whether tolerations are added only once. Since this was the initial issue, maybe this deserves an e2e. Correct me if I'm wrong.

Reply:
Good point, this should actually check if the tolerations are identical since only the pods get the default kubernetes tolerations added to them so it's not necessary to iterate and check if they are contained.